nx/lib/nx/defn/expr.ex

```diff
@@ -230,16 +251,61 @@ defmodule Nx.Defn.Expr do
     size = tuple_size(head)
     expr = fun.(tuple_out(size))
 
-    {out, {[], ^size}} =
-      Composite.traverse(out, {Tuple.to_list(head), 0}, fn _, {[head | tail], i} ->
-        {expr(head, context, :elem, [expr, i]), {tail, i + 1}}
+    vectorized_axes =
+      out
+      |> Composite.reduce([], fn
+        %T{vectorized_axes: [_ | _] = axes}, acc ->
+          [Keyword.keys(axes) | acc]
+
+        _t, acc ->
+          [nil | acc]
```

Comment:
If we use an empty list here, I think we can skip the conditional in the traverse below.

```suggestion
          [[] | acc]
```

Reply:
We can't. There's a check for scalars in vectorize that impedes this.

Reply:
Ah.